Public/Src/Cache/ContentStore/Distributed/NuCache/RocksDbContentLocationDatabase.cs

```diff
@@ -90,13 +91,28 @@ public RocksDbContentLocationDatabase(IClock clock, RocksDbContentLocationDataba
         /// <inheritdoc />
         protected override Task<BoolResult> ShutdownCoreAsync(OperationContext context)
         {
+            lock (this)
+            {
+                _compactionTimer?.Dispose();
+            }
+
             _keyValueStore?.Dispose();
+
             return base.ShutdownCoreAsync(context);
         }
 
         /// <inheritdoc />
         protected override BoolResult InitializeCore(OperationContext context)
         {
+            if (_configuration.FullRangeCompactionInterval != Timeout.InfiniteTimeSpan)
+            {
+                _compactionTimer = new Timer(
+                    _ => FullRangeCompaction(context),

```

Comment:
Should we add a logger for when this fails?
Not sure if the timer kills the entire process, if the timer stops functioning on errors and not perform cleanup anymore, etc...